pkg/adsc/adsc.go

```diff
@@ -551,12 +522,10 @@ func (a *ADSC) handleRecv() {
 
 		a.mutex.Lock()
 		if len(gvk) == 3 {
-			gt := resource.GroupVersionKind{gvk[0], gvk[1], gvk[2]}
+			gt := resource.GroupVersionKind{Group: gvk[0], Version: gvk[1], Kind: gvk[2]}
 			a.sync[gt.String()] = time.Now()
 		}
 		a.Received[msg.TypeUrl] = msg
-		// TODO: add hook to inject nacks
-		a.ack(msg)
```

Comment:
where is the ACK now? It seems missing entirely?

Reply:
Ack is not needed when you respond with next request as is being done below.

Reply:
According to what?

> The xDS client should ACK or NACK every DiscoveryResponse received from the management server.

And I confirmed that Envoy behaves this way

Even if our tests pass we probably want to simulate envoy as closely as possible?

Reply:
https://www.envoyproxy.io/docs/envoy/latest/api-docs/xds_protocol#aggregated-discovery-service - For ADS Ack is the next EDS request for CDS response.

Reply:
I can change this to send ACK for EDS/RDS to match exactly what Envoy expects in ADS. But for CDS/LDS - Envoy follows it with an EDS request/RDS request